### PR TITLE
[Dashboard] Add serve controller info to the Serve page 

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
Screenshot 2023-05-14 at 4 58 26 PM

Screenshot 2023-05-14 at 4 56 47 PM

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
We can probably remove the actor logs for now?

File: dashboard/client/src/components/StatusChip.tsx [hunk 1]
```diff
@@ -85,9 +85,14 @@ const colorMap = {
     [ServeReplicaState.STOPPING]: red,
   },
   serveHttpProxy: {
-    [ServeHTTPProxyStatus.HEALTHY]: green,
-    [ServeHTTPProxyStatus.UNHEALTHY]: red,
-    [ServeHTTPProxyStatus.STARTING]: yellow,
+    [ServeSystemActorStatus.HEALTHY]: green,
```
Comment (galenhwang) [inline dashboard/client/src/components/StatusChip.tsx:88]: nit: future TODO, type keys as enum or `const x = [] as const; type xType = typeof x[number];`

File: dashboard/client/src/pages/serve/hook/useServeApplications.ts [hunk 2]
```diff
@@ -206,10 +212,32 @@ export const useServeHTTPProxyDetails = (httpProxyId: string | undefined) => {
...
+    "useServeControllerDetails",
+    async () => {
+      const rsp = await getServeApplications();
+
+      if (rsp) {
+        return rsp.data;
+      }
+    },
+    { refreshInterval: API_REFRESH_INTERVAL_MS },
+  );
```
Comment (galenhwang) [inline dashboard/client/src/pages/serve/hook/useServeApplications.ts:234]: nit: this should be extracted to a general `use(Get/Fetch)ServeApplications` hook, i see it repeated a fair bit

File: dashboard/client/src/type/serve.ts [hunk 3]
```diff
@@ -86,22 +86,25 @@ export enum ServeDeploymentMode {
...
+export type ServeSystemActor = {
+  node_id: string | null;
+  node_ip: string | null;
+  actor_id: string | null;
+  actor_name: string | null;
+  worker_id: string | null;
   log_file_path: string | null;
 };
 
+export type ServeHttpProxy = {
```
Comment (galenhwang) [inline dashboard/client/src/type/serve.ts:104]: nit/personal pref: i would put inherited types first before extended: 
```
type B = A & {
...
};

**Review by galenhwang — APPROVED** [review]

**Comment by galenhwang** [comment]
(unrelated to this PR but posting here)

is showing "Deploying app <app> failed" while the serve app is running expected? I tried to deploy initially and failed, which showed `Status: DEPLOY_FAILED` I believe. On redeploy after changing some configs, it shows `RUNNING`, but the status message remains

Screenshot 2023-05-15 at 11 31 10 AM

File: dashboard/client/src/pages/serve/ServeSystemActorDetailPage.tsx [hunk 4]
```diff
@@ -38,27 +41,93 @@ export const ServeHttpProxyDetailPage = () => {
...
+  | {
+      type: "controller";
+      detail: ServeSystemActor;
+    };
+
+type ServeSystemActorDetailProps = {
+  actor: ActorInfo;
+};
+
+export const convertActorStateForServeController = (
```
Comment (galenhwang) [inline dashboard/client/src/pages/serve/ServeSystemActorDetailPage.tsx:112]: nit: use Record/obj instead of func

**Comment by alanwguo** [comment]
> (unrelated to this PR but posting here)
> 
> is showing "Deploying app failed" status message while the serve app is running expected? I tried to deploy initially and failed, which showed `Status: DEPLOY_FAILED` I believe with the failed status message. On redeploy after changing some configs, it shows `RUNNING`, but the status message remains
> 
> Screenshot 2023-05-15 at 11 31 10 AM

@zcin is this expected?

@galenhwang , for me, the message usually clears out after I fix the deployment...

File: dashboard/client/src/pages/serve/ServeSystemActorDetailPage.tsx (hunk 4 above)
Comment (alanwguo) [reply dashboard/client/src/pages/serve/ServeSystemActorDetailPage.tsx:112]: any reason for this? With the function approach, I can easily add a "catch-all" by doing an `else` block at the end.

**Comment by rkooo567** [comment]
There's merge conflict

**Comment by zcin** [comment]
> @zcin is this expected?

Sorry for the delayed response .. this is not expected, @galenhwang can you file an issue?

**Pull request merged**